src/storage/data_table.cpp

```diff
@@ -180,24 +225,65 @@ bool DataTable::ScanBaseTable(Transaction &transaction, DataChunk &result, Table
 		current_row += STANDARD_VECTOR_SIZE;
 		return true;
 	}
-
-	for (idx_t i = 0; i < state.column_ids.size(); i++) {
-		auto column = state.column_ids[i];
-		if (column == COLUMN_IDENTIFIER_ROW_ID) {
-			// scan row id
-			assert(result.data[i].type == ROW_TYPE);
-			result.data[i].Sequence(base_row + current_row, 1);
+	//! If we have filters
+	if (!table_filters.empty() && !apply_filter) {
+		SelectionVector sel;
+		idx_t approved_tuple_count = count;
+		if (count != max_count) {
+			sel = valid_sel;
 		} else {
-			// scan actual base column
-			columns[column].Scan(transaction, state.column_scans[i], result.data[i]);
+			sel = FlatVector::IncrementalSelectionVector;
+		}
+		//! First, we scan the columns with filters, fetch their data and generate a selection vector.
+		for (auto &table_filter : table_filters) {
+			apply_filter = columns[table_filter.first].Select(transaction, state.column_scans[table_filter.first],
+			                                                  result.data[table_filter.first], sel,
+			                                                  approved_tuple_count, table_filter.second);
+			if (apply_filter) {
+				break;
+			}
+		}
+		if (!apply_filter) {
+			result.Slice(sel, approved_tuple_count);
```

Comment:
Here you only want to slice the vectors that you have filtered, right? Rather than slicing the entire chunk I would slice ONLY those vectors. Slicing the entire chunk could cause problems.